### DISCUSSION TITLE
Add Id/Uid property to interactive objects

### DISCUSSION THREAD

**Discussion by fibiaanm** [fabricjs/fabric.js#10090] [Ideas]
Hello!

I've been using fabric for 3 months (I have enjoyed it so much). I am using it for a designer project and I extended the Object classes to add the "id" property (it's really an uuid). That has helped me to identify objects, to interact with them more easily, and to create a version control system (undo/redo feature). I am a big fan of uuid, but instead of numeric IDs for this kind of work.

I think it will be a useful property for the devs. I've been doing some research and the property could be added to the InteractiveFabricObject class, and that property will be easily modified with the "_set" method in the CommonMethods Class. This could also help us to implement a system in the future. 

If you find this feature good enough, I'd be glad to develop it.

Thank you in advance!

**Comment by asturur** [comment]
Hello @fibiaanm with the latest version of fabricJS supporting uuid should boil down to:

- make an uuid ( each object )
- pass it in the object contructor ( each object )
- modify the static object property `customProperties` to ['id'] or ['uuid'] (once)
- modify the typescript interface to explain TS that id or uuid is a legit field (once)

It should be described in this doc here:
https://fabricjs.github.io/docs/using-custom-properties/

Is there any reason why you think it should get some kind of internal support?
